src/scores/contingency.py

```diff
@@ -0,0 +1,258 @@
+		self.fn_count = self.fn.sum().values.item()     # Count of true negatives
+		self.total_count = self.tp_count + self.tn_count + self.fp_count + self.fn_count
+
+	def accuracy(self) -> float:
+		'''
+		The proportion of forecasts which are true
+		'''
+
+		correct_count = self.tp_count + self.tn_count
+		ratio = correct_count / self.total_count
+		return ratio
+
+	def frequency_bias(self) -> float:
+		'''
+		'''
+		freq_bias = (self.tp_count + self.fp_count) / (self.tp_count + self.fn_count)
+		return freq_bias
+
+	def probability_of_detection(self) -> float:
+		'''
+		What proportion of the observed events where correctly forecast?
+		Range: 0 to 1.  Perfect score: 1.
+		'''
+
+		pod = self.tp_count / (self.tp_count + self.fn_count)
+		return pod
+
+	def false_alarm_rate(self) -> float:
+		'''
+		What fraction of the non-events were incorrectly predicted?
+		Range: 0 to 1.  Perfect score: 0.
+		'''
+
+		far = self.fp_count / (self.tn_count + self.fp_count)
+		return far
+
+	def success_ratio(self) -> float:
+		'''
+		What proportion of the forecast events actually eventuated?
+		Range: 0 to 1.  Perfect score: 1.
+		'''
+
+		sr = self.tp_count / (self.tp_count + self.fp_count)
+		return sr
+
+	def threat_score(self) -> float:
+		'''
+		How well did the forecast "yes" events correspond to the observed "yes" events?
+		Range: 0 to 1, 0 indicates no skill. Perfect score: 1.
+		'''
+		ts = self.tp_count / (self.tp_count + self.fp_count + self.tn_count)
+		return ts
+
+	def sensitivity(self) -> float:
+		'''
+		What proportion of non-events were correctly predicted?
+		'''
+
+		s = self.tn_count / (self.tn_count + self.fp_count)
+		return s
+
+
+
+
+
+class BinaryTable():	
+	'''
+	At each location, the value will either be:
+	 - True (Accurate)
+	 - False (Not Accurate)
+
+	It will be common to want to operate on masks of these values,
+	such as:
+	 - Plotting these attributes on a map
+	 - Calculating the total number of these attributes
+	 - Calculating various ratios of these attributes, potentially
+	   masked by geographical area (e.g. accuracy in a region)
+
+	As such, the per-pixel information is useful as well as the overall
+	ratios involved.	
+	'''
+
+	def __init__(self, data):
+
+		bool_array = data.astype(bool)
+		self.table = xarray.DataArray(bool_array)
+			
+	def hits(self):
+		'''
+		The 'hits' on a binary table is simply the sum of the table
+		'''
+		total = self.table.sum().values.item()
+		return total
+	
+
+class MatchingOperator:
+	pass
+
+class EventThresholdOperator(MatchingOperator):
+	'''
+	Given a forecast and and an observation, consider an event defined by
+	particular variables meeting a threshold condition (e.g. rainfall above 1mm).
+
+	This class abstracts that concept for any event definition.
+	'''
+
+	def __init__(self, *, precision=DEFAULT_PRECISION, tolerance=0):
+		self.precision = precision
+		self.tolerance = tolerance
+
+	def make_table(self, forecast, observed, threshold, op_fn=operator.gt):
```

Comment:
Don't hardcode `op_fn=operator.gt`. Allow the user to choose the operator

Reply:
It's not hard-coded, just provided as a default. The user could supply any suitable of function.